kernel/kernel-api/src/test/scala/io/delta/kernel/internal/columndefaults/ColumnDefaultsSuite.scala

```diff
@@ -0,0 +1,166 @@
+      .add("long", LongType.LONG, metadataForDefault("1231341"))
+      .add("decimal", new DecimalType(10, 5), metadataForDefault("1231.34155"))
+      .add("float", FloatType.FLOAT, metadataForDefault("123.1341"))
+      .add("double", DoubleType.DOUBLE, metadataForDefault("123.7774"))
+      .add("double2", DoubleType.DOUBLE, metadataForDefault("'123.7774'"))
+      .add("name", StringType.STRING, metadataForDefault("\"tom\""))
+      .add("name2", StringType.STRING, metadataForDefault("'tom'"))
+      .add("date", DateType.DATE, metadataForDefault("'2025-01-01'"))
+      .add("date2", DateType.DATE, metadataForDefault("2025-01-01"))
+      .add("ts", TimestampType.TIMESTAMP, metadataForDefault("\"2025-01-01T00:00:00Z\""))
+      .add("ts2", TimestampType.TIMESTAMP, metadataForDefault("\"2025-01-01T00:00:00+01:00\""))
+      .add("ts3", TimestampType.TIMESTAMP, metadataForDefault("2025-01-01T00:00:00Z"))
+      .add("tsntz", TimestampNTZType.TIMESTAMP_NTZ, metadataForDefault("'2025-01-01T00:00:00'"))
+      .add("tsntz2", TimestampNTZType.TIMESTAMP_NTZ, metadataForDefault("2025-01-01T00:00:00"))
+      .add(
+        "childStruct",
+        new StructType()
+          .add("childId", IntegerType.INTEGER, metadataForDefault("100"))
+          .add(
+            "grandChildList",
+            new ArrayType(
+              new StructType().add("nestedId", IntegerType.INTEGER, metadataForDefault("120")),
+              false)))
+      .add(
+        "grandChildMap",
+        new MapType(
+          new StructType().add("mapKeyId", IntegerType.INTEGER, metadataForDefault("220")),
+          new StructType().add("mapValueId", IntegerType.INTEGER, metadataForDefault("330")),
+          false))
+      .add(
+        "childList",
+        new ArrayType(
+          new StructType().add("clid", IntegerType.INTEGER, metadataForDefault("300")),
+          false))
+    ColumnDefaults.validateSchema(correctSchema, true, true)
+    val e = intercept[KernelException] {
+      ColumnDefaults.validateSchema(correctSchema, true, false)
+    }
+    assert(e.getMessage ==
+      "In Delta Kernel, default values table feature requires IcebergCompatV3 to be enabled.")
+    val e2 = intercept[KernelException] {
+      ColumnDefaults.validateSchema(correctSchema, false, true)
+    }
+    assert(e2.getMessage ==
+      "This table does not enable table features for setting column defaults")
+
+    val unsupportedCases = Seq(
+      new StructType().add("sub", IntegerType.INTEGER),
+      new ArrayType(IntegerType.INTEGER, false),
+      new MapType(IntegerType.INTEGER, IntegerType.INTEGER, false),
+      VariantType.VARIANT)
+    unsupportedCases.foreach { dataType =>
+      val schemaWithUnsupportedType = new StructType()
+        .add("id", IntegerType.INTEGER)
+        .add("col1", dataType, metadataForDefault("120"))
+      intercept[KernelException] {
+        ColumnDefaults.validateSchema(schemaWithUnsupportedType, true, true)
+      }
```

Comment:
check error message here -- as outlined above, I think this should be specific as to the issue (unsupported type vs not a literal expression)